File: benchmark/c/cpw/write_one_line_c_function_to_find_whether_a_no_is_power_of_two/write_one_line_c_function_to_find_whether_a_no_is_power_of_two.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int write_one_line_c_function_to_find_whether_a_no_is_power_of_two ( int n ) ;
int write_one_line_c_function_to_find_whether_a_no_is_power_of_two ( int n ) {
  if ( n == 0 ) return 0;
  while ( n != 1 ) {
    if ( n % 2 != 0 ) return 0;
    n = n / 2;
  }
  return 1;
}

File: translation_result/else_bms/success/c/cpw/write_one_line_c_function_to_find_whether_a_no_is_power_of_two/gpt4/Restart/write_one_line_c_function_to_find_whether_a_no_is_power_of_two.rs

fn write_one_line_c_function_to_find_whether_a_no_is_power_of_two(mut n: i32) -> bool {
    if n == 0 {
        return false;
    }
    while n != 1 {
        if n % 2 != 0 {
            return false;
        }
        n = n / 2;
    }
    true
}
